### PR TITLE
Update slug on displayName update

### PR DESCRIPTION
**Pull request by jpaddison3**
lesswrongteam let me know if you don't want this and I'll forum gate it. My argument in favor: most users who want to change their displayName also want to change their slug, many of them pretty strongly. Although I think it's reasonably to have a "formerly known as" field or something, I don't think using the slug is the right way to do it, and it should probably not be mandatory.

This PR also prevents admins from accidentally adding upper case letters to slugs, which is how I got started on this rabbit hole.

This leads to an error on the resetStore call, as apollo's attempt to rerun the `{slug: 'outdatedSlug'}` query will now fail. I decided I didn't mind the rare console error here, but I could be persuaded to find a more elegant solution. (I can't find anything in the apollo docs about removing a _query_ from the store. I could try to modify that query on the mutation result, but my attempts to do similar things in the past have been a pain, and the upside here is low.)

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1202654344329585) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by darkruby501** [comment]
This has been on my to-do list for a while. Just double-checking, when the slug is updated, the old one still works, right?

**Comment by jpaddison3** [comment]
Yep. See the logic about oldSlugs.

File: packages/lesswrong/server/vulcan-lib/mutators.ts [hunk 1]
```diff
@@ -326,6 +329,7 @@ export const updateMutator = async <T extends DbObject>({
 
   */
   if (validate) {
+    logger('vadidating')
```
Comment (oetherington) [inline packages/lesswrong/server/vulcan-lib/mutators.ts:332]: Small typo

**Review by oetherington — APPROVED** [review]
Sorry for the slightly slow review. When I tested it out I got a really weird error that I wanted to track down, but it turns out I just messed up my user in the database whilst working on a different ticket.

Seems to work nicely!

**Pull request merged**